docs/overview.rst

```diff
@@ -15,7 +414,7 @@ sight, but can be very powerful.
    IterationScheme -> RequestIterator [label=" Returns"];
    RequestIterator -> DataIterator [style=dashed, label=" Argument to"];
    DataIterator -> DataStream [label=" Gets data from"];
-   DataStream -> DataStream [style=dashed, label=" Gets data from (wrapper)"];
+   DataStream -> DataStream [style=dashed, label=" Gets data from (transformer)"];
    { rank=same; RequestIterator DataIterator }
 
 Datasets
```

Comment:
I would move the visual diagram at the end of the `Division of labour` section.

The textual summary instead doesn't seem to be useful anymore and, on the contrary, seems to be either wrong or at least badly phrased in some parts, e.g.: 
> Data streams can produce data set iterators (epoch iterators) which will use the data stream's state to return actual data.

My understanding is that 
* `Iterators` produce list of indexes 
* `DataStream`s do not produce `Iterators`, they are *given* a `Dataset` and an `Iterator` and produce actual data.

As such, I suggest to completely remove it.